tests/cylc-get-config/05-param-vars.t

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2017 NIWA
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+# Test parameter vars are not defined with user env GitHub #2225
+
+. "$(dirname "$0")/test_header"
+
+set_test_number 2
+
+install_suite "${TEST_NAME_BASE}" "${TEST_NAME_BASE}"
+
+TEST_NAME="${TEST_NAME_BASE}-config"
+run_ok "${TEST_NAME}" \
+    cylc get-config -i "[runtime][foo_t1_right]environment" "${SUITE_NAME}"
+cmp_ok "${TEST_NAME}.stdout" - <<__END__
+PARAM1 = \$CYLC_TASK_PARAM_t
+PARAM2 = \$CYLC_TASK_PARAM_u
+__END__
+
+purge_suite "${SUITE_NAME}"
```

Comment:
It occurs to me this test correctly determines that the parameter values are no longer set in the user-defined task environment, but it doesn't test that they are correctly set in the suite-defined task environment.  Can you try adding several more tests to this test script:
 * generate a job script via `run_ok` using `cylc jobscript ${SUITE_NAME}  foo_t1_right.1`
 * then use `grep_ok` to check that `$CYLC_TASK_PARAM_t` and `..._u` are correctly defined in it.

(An alternative would be to run a test suite in which the task foo_t1_right uses its parameter variables at run time - however, checking the job script will do in this case).